### DISCUSSION TITLE
Librdkafka integration on embedded device

### DISCUSSION THREAD

**Discussion by nikidimitrow** [edenhill/librdkafka#3832] [Q&A] [answered]
Hello support team,

I'm try to run [librdkafka](https://github.com/edenhill/librdkafka) (as a Kafka consumer) on the microcontroller ESP32 (https://www.espressif.com/en/products/socs/esp32) 

As microcontroller are well know for small flash memory and CPU power compare with desktop PC my questions are following:

- Is there a version of librdkafka for embedded device?
- Some bare minimal and fully strip from desktop variants of the project?
- Somebody already did something similar?
- In your opinion is it possible to be perform integration in librdkafka on embedded device?

Excuse me for so general questions.

Have a great evening and cold beers.

**Comment by edenhill** [comment, marked as answer]
librdkafka requires posix threads, multiple cores, and megabytes of memory, so it will not run on an ESP32 or similar.
I recommend using a simple MQTT or HTTP client on the ESP32 and an MQTT connector or the Kafka REST proxy on the serer side to bridge between the embedded devices and Kafka.

**Reply by franz-ms-muc** [reply]
Posix Threads are available. 
Many Megabytes of RAM not .

**Comment by nikidimitrow** [comment]
Espressif had confirm the statement of @edenhill, that integration of librdkafka is not possible on IoT device: https://github.com/espressif/esp-idf/issues/8848#issuecomment-1112015022
